### PR TITLE
Refactor developing-dapr documents for consistency

### PR DESCRIPTION
**Pull request by CodeMonkeyLeet**
# Description

Address consistency issues mentioned in #2529:
- Consolidate all dapr development environment setup instructions in setup-dapr-development-env.md and remove duplication from developing-dapr.md.
- Add golangci-lint as a prerequisite for `make lint`.
- Add linux build-essentials package as requirement for `make` & `gcc`.
- Remove incorrect Windows instruction to set `GOPATH=C:\go`. `GOPATH` should default to `%USERPROFILE%\go` as indicated by golang install instructions.
- Add links to dapr docs for debugging dapr in Kubernetes.

A separate PR will address the remaining concerns regarding the use of mingw and WSL2 for Windows dev environments.

## Issue reference

Addresses most concerns in #2529, requires additional PR to close on mingw32-make issues on Windows.

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: docs/development/setup-dapr-development-env.md [hunk 1]
```diff
@@ -1,57 +1,91 @@
...
+1. Download and install [Go 1.16 or later](https://golang.org/doc/install#tarball).
+   - Make sure that your `GOPATH` and `PATH` environment variables are configured so that gotest modules can be run. For example:
+
    ```bash
    export GOPATH=~/go
    export PATH=$PATH:$GOPATH/bin
    ```
-2. [Delve](https://github.com/go-delve/delve/tree/master/Documentation/installation) for Debugging
+
+2. Install the [Delve](https://github.com/go-delve/delve/tree/master/Documentation/installation) debugger for Go.
```
Comment (daixiang0) [inline docs/development/setup-dapr-development-env.md:24]: mark it optional?

File: docs/development/setup-dapr-development-env.md [hunk 2]
```diff
@@ -1,57 +1,91 @@
...
+
+3. Install [golangci-lint](https://golangci-lint.run/usage/install).
 
 ### Windows
 
 1. The Go language environment [(instructions)](https://golang.org/doc/install#windows).
```
Comment (daixiang0) [inline docs/development/setup-dapr-development-env.md:30]: keep same format with Linux:
```Download and install [Go 1.16 or later]```

File: docs/development/setup-dapr-development-env.md [hunk 3]
```diff
@@ -1,57 +1,91 @@
...
     ```
 
-## Setup Kubernetes development environment
+## Setup a Kubernetes development environment
+
+Follow the guide on [how to set up a Kubernetes cluster for Dapr](https://docs.dapr.io/operations/hosting/kubernetes/cluster/).
+
+For development purposes, you will also want to follow the optional steps to install [Helm 3.x](https://helm.sh/docs/intro/install/).
+
+### Installing Make
```
Comment (daixiang0) [inline docs/development/setup-dapr-development-env.md:67]: For no-k8s env, it still needs.

Comment (CodeMonkeyLeet) [reply docs/development/setup-dapr-development-env.md]: Thanks for the catch, this was incorrectly nested under the previous header!

File: docs/development/setup-dapr-development-env.md (hunk 2 above)
Comment (CodeMonkeyLeet) [reply docs/development/setup-dapr-development-env.md]: I fixed up the wording to be more consistent with the Linux side, but I wouldn't get too hung up on this for now; as suggested in #2529, the Windows instructions are going to be reworked more in an upcoming PR anyway.

File: docs/development/setup-dapr-development-env.md (hunk 1 above)
Comment (CodeMonkeyLeet) [reply docs/development/setup-dapr-development-env.md]: Added an indication for user choice.

File: docs/development/developing-dapr.md [hunk 4]
```diff
@@ -1,81 +1,12 @@
...
-
-1. The Go language environment [(instructions)](https://golang.org/doc/install#tarball).
-   * Make sure that your GOPATH and PATH are configured correctly
-   ```bash
-   export GOPATH=~/go
-   export PATH=$PATH:$GOPATH/bin
-   ```
-2. [Delve](https://github.com/go-delve/delve/tree/master/Documentation/installation) for Debugging
-
-#### Windows
```
Comment (artursouza) [inline docs/development/developing-dapr.md:31]: I actually use this configuration. Why are we deleting these instructions?

**Review by artursouza — CHANGES_REQUESTED** [review]

File: docs/development/developing-dapr.md (hunk 4 above)
Comment (CodeMonkeyLeet) [reply docs/development/developing-dapr.md:31]: They are duplicated in the setup-dapr-development-env.md, with slight variations that caused the confusion in #2529. This PR just consolidates these instructions into setup-dapr-development-env.md and removes the duplication in developing-dapr.md, which now points to the other doc for manual install steps.

**Review by artursouza — APPROVED** [review]

**Comment by artursouza** [comment]
@CodeMonkeyLeet Please, comment on https://github.com/dapr/dapr/issues/2529 so it can be assigned to you.

**Pull request merged**